### PR TITLE
Fix bug 1178252 - Stop updated the json blob when a wiki document is logically deleted like when it's marked as spam.

### PR DESCRIPTION
**Pull request by jezdez**
This also fixes and reenables some deferred tests that were skipped.

### CODE REVIEW

File: kuma/wiki/tests/test_models.py [hunk 1]
```diff
@@ -891,43 +891,61 @@ def test_get_rendered(self, mock_kumascript_get):
...
         # Initially empty json field should be filled in after render()
-        eq_(None, self.d1.json)
+        eq_(self.d1.json, None)
         self.d1.render()
+        # reloading from db to get the updates done in the celery task
+        self.d1 = Document.objects.get(pk=self.d1.pk)
         ok_(self.d1.json is not None)
 
-        time.sleep(0.1)  # Small clock-tick to age the results.
+        time.sleep(1.0)  # Small clock-tick to age the results.
```
Comment (jezdez) [inline kuma/wiki/tests/test_models.py]: This was needed for the test to pass :(

File: kuma/wiki/tests/test_models.py [hunk 2]
```diff
@@ -891,43 +891,61 @@ def test_get_rendered(self, mock_kumascript_get):
...
+        """
         config.KUMASCRIPT_TIMEOUT = 1.0
         mock_kumascript_get.return_value = ('<p>summary!</p>', None)
-
         ok_(not self.d1.rendered_html)
         result_summary = self.d1.get_summary()
+        ok_(not mock_kumascript_get.called)
+        ok_(not self.d1.rendered_html)
+
+        self.d1.render()
```
Comment (jezdez) [inline kuma/wiki/tests/test_models.py]: This is needed now that get_summary doesn't call get_rendered anymore.

File: kuma/wiki/tests/test_models.py (hunk 1 above)
Comment (kleberveridiano) [inline kuma/wiki/tests/test_models.py]: ok!

File: kuma/wiki/tests/test_models.py (hunk 2 above)
Comment (kleberveridiano) [inline kuma/wiki/tests/test_models.py]: ok!

**Comment by kleberveridiano** [comment]
I´m very happy now!

File: kuma/wiki/tests/test_models.py (hunk 1 above)
Comment (groovecoder) [inline kuma/wiki/tests/test_models.py]: :cry:

File: kuma/wiki/tests/test_models.py (hunk 2 above)
Comment (groovecoder) [inline kuma/wiki/tests/test_models.py]: :+1:

**Pull request merged**